src/test/java/com/beust/jcommander/parameterized/parser/JsonAnnotationParameterizedParser.java

```diff
@@ -0,0 +1,256 @@
+          }
+        }
+      }
+
+      /*
+      * This section would be for completeness and although it is not tested it is left here
+      * as a template to use the JsonSetter (or JsonGetter) methods as ways to define parameters
+      * at runtime.
+      */
+//      // check methods
+//      for (Method method : curClazz.getDeclaredMethods()) {
+//        // these only work on setMethods
+//        if (!method.getName().startsWith("set")) {
+//          continue;
+//        }
+//
+//        JsonSetter jsonSetterAnnotation = (JsonSetter) method.getAnnotation(JsonSetter.class);
+//        if (jsonSetterAnnotation != null) {
+//          Map<String, Object> map = new HashMap<String, Object>();
+//
+//          /*
+//          * For primitive and their derived types, we can use the Parameter annotation, but for
+//          * other user classes, we need to add a delegate
+//           */
+// /*
+//            * create standard Parameter
+//           */
+//          String name = jsonSetterAnnotation.value();
+//          map.put("names", new String[]{name});
+//          map.put("required", false);
+//          //
+//          // TODO SET THE DEFAULT VALUE BASED ON THE values() OR valuesEnum()
+//          // map.put("default", annotation.defaultValue());
+//          //
+//          map.put("descriptionKey", "");
+//          // get the parameter type
+//          Class[] paramTypes = method.getParameterTypes();
+//          // there should only be one for a 
+//          // all variable types, even Boolean require 1 following parameter.
+//          //if (paramTypes[0] == Boolean.class || paramTypes[0] == boolean.class) {
+//          map.put("arity", 1);
+//          //} 
+//          map.put("variableArity", (paramTypes[0] == List.class));
+//          map.put("password", false);
+//          map.put("converter", NoConverter.class);
+//          map.put("listConverter", NoConverter.class);
+//          map.put("hidden", false);
+//          map.put("validateWith", new Class[]{NoValidator.class});
+//          map.put("validateValueWith", new Class[]{NoValueValidator.class});
+//          map.put("splitter", CommaParameterSplitter.class);
+//          map.put("echoInput", true);
+//          map.put("help", false);
+//          map.put("forceNonOverwritable", false);
+//          map.put("order", -1);
+//          map.put("description", "");
+//
+//          Parameter param = (Parameter) AnnotationParser.annotationForMap(Parameter.class, map);
+//          result.add(new Parameterized(new WrappedParameter(param), null, null, method));
+//        }
+//      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Basic check for primitive or Java class that should be used directly.
+   * 
+   * @param field
+   * @return true if Java primitive or part of the Java or Sun package.
+   */
+  public boolean isPrimitiveOrString(Field field) {
+    final Class type = field.getType();

```

Comment:
No need for `final`.

Also, isn't the compiler complaining about the raw `Class` types?